user-service/services/user_service.go

```
@@ -136,16 +134,19 @@ func (s *UserService) Login(ctx context.Context, loginDto *dtos.UserLoginDto) er
 
 // CreateNewToken issues a signed JWT for the authenticated user.
 func (s *UserService) CreateNewToken(ctx context.Context, user *entities.User) (string, error) {
-	token := jwt.NewWithClaims(jwt.SigningMethodHS256, jwt.MapClaims{
+	token := jwt.NewWithClaims(jwt.SigningMethodRS256, jwt.MapClaims{
 		"sub":      user.ID,
 		"name":     strings.Join([]string{user.FirstName, user.LastName}, " "),
 		"username": user.Username,
 		"role":     user.Role,
+		"status":   user.AccountStatus,
 		"iat":      time.Now().Unix(),
 		"exp":      time.Now().Add(15 * time.Minute).Unix(),
 	})
 
-	tokenString, err := token.SignedString(hmacSampleSecret)
+	pk, _ := utils.GetPrivateKey()
```

This should return an error rather than ignoring it.